kornia/feature/responses.py

```diff
@@ -94,7 +94,7 @@ def harris_response(
     if sigmas is not None:
         scores = scores * sigmas.pow(4).view(-1, 1, 1, 1)
 
-    return scores
+    return scores.type_as(input)
```

Comment:
at which point the dtype start to diverge ? seems al related to spatial gradient ?

Reply:
Again it's the .pow(4) which does it here

Reply:
If we replace pow(4) with 4 multiplications hardcoded, would it work?

Reply:
Is this the type that changes because of the pow or does sigma have a different type?
In general, PyTorch doesn't upcase, maybe if you use AMP (when it would for good reasons).

```
torch.pow(torch.randn(1, device="cuda", dtype=torch.float16), 4)
tensor([0.0247], device='cuda:0', dtype=torch.float16)
```